client/jetpack-cloud/sections/partner-portal/payment-method-delete-primary-confirmation/index.tsx

```diff
@@ -0,0 +1,85 @@
+import { PaymentLogo } from '@automattic/composite-checkout';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect } from 'react';
+import { QueryKey, useQuery } from 'react-query';
+import { wpcomJetpackLicensing as wpcomJpl } from 'calypso/lib/wp';
+import type { PaymentMethod } from 'calypso/jetpack-cloud/sections/partner-portal/payment-methods';
+import type { Dispatch, FunctionComponent, SetStateAction } from 'react';
+
+import './style.scss';
+
+const getCacheKey = ( key: string ): QueryKey => [ 'jetpack-cloud', 'partner-portal', key ];
+
+interface Props {
+	paymentMethod: PaymentMethod;
+	setNextPrimaryPaymentMethod: Dispatch< SetStateAction< PaymentMethod | null > >;
+}
+
+const PaymentMethodDeletePrimaryConfirmation: FunctionComponent< Props > = ( {
+	paymentMethod,
+	setNextPrimaryPaymentMethod,
+} ) => {
+	const translate = useTranslate();
+
+	const { data: recentCards, isFetching } = useQuery( getCacheKey( 'recent-cards' ), () =>
+		wpcomJpl.req.get( {
+			apiNamespace: 'wpcom/v2',
+			path: '/jetpack-licensing/stripe/payment-methods',
+		} )
+	);
+
+	const nextPrimaryPaymentMethod = ( recentCards?.items || [] ).find(
+		( card: PaymentMethod ) => card.id !== paymentMethod.id
+	);
+
+	useEffect( () => {
+		setNextPrimaryPaymentMethod( nextPrimaryPaymentMethod );
+	}, [ nextPrimaryPaymentMethod, setNextPrimaryPaymentMethod ] );
+
+	if ( ! isFetching && ! nextPrimaryPaymentMethod ) {
+		return <></>;
+	}
```

Comment:
In a case like this, I prefer to return `null` rather than an empty component (fragment), which React has to process/create. From the consumer perspective, the return value will be `null` instead of a component if these conditions are met.

Reply:
Thanks! Fixed in 0aed5538d432fafba6335a497f2fb43d58a43ef1